lib/src/feedback_bottom_sheet.dart

```diff
@@ -1,96 +1,38 @@
-import 'package:feedback/src/l18n/translation.dart';
+import 'package:feedback/src/better_feedback.dart';
 import 'package:feedback/src/theme/feedback_theme.dart';
 import 'package:flutter/material.dart';
 
-typedef OnSubmit = void Function(BuildContext context, String feedback);
-
 /// Shows the text input in which the user can describe his feedback.
 class FeedbackBottomSheet extends StatelessWidget {
-  const FeedbackBottomSheet({Key? key, required this.onSubmit})
-      : super(key: key);
-
-  final OnSubmit onSubmit;
-
-  @override
-  Widget build(BuildContext context) {
-    return Overlay(
-      initialEntries: [
-        OverlayEntry(
-          builder: (context) {
-            return _FeedbackBottomSheet(
-              onSubmit: onSubmit,
-            );
-          },
-        ),
-      ],
-    );
-  }
-}
-
-class _FeedbackBottomSheet extends StatefulWidget {
-  const _FeedbackBottomSheet({
+  const FeedbackBottomSheet({
     Key? key,
+    required this.getFeedback,
     required this.onSubmit,
   }) : super(key: key);
 
+  final GetFeedback getFeedback;
   final OnSubmit onSubmit;
 
-  @override
-  __FeedbackBottomSheetState createState() => __FeedbackBottomSheetState();
-}
-
-class __FeedbackBottomSheetState extends State<_FeedbackBottomSheet> {
-  late TextEditingController controller;
-
-  @override
-  void initState() {
-    super.initState();
-    controller = TextEditingController();
-  }
-
   @override
   Widget build(BuildContext context) {
-    return Column(
-      mainAxisSize: MainAxisSize.min,
-      mainAxisAlignment: MainAxisAlignment.end,
-      children: [
-        ColoredBox(
+    // We need to supply a navigator so that the contents of the bottom sheet
+    // have access to an overlay (overlays are used by many material widgets
+    // such as `TextField` and `DropDownButton`.
+    // Typically, the navigator would be provided by a `MaterialApp`, but
+    // `BetterFeedback` is used above `MaterialApp` in the widget tree so that
+    // the nested navigation in navigate mode works properly.
+    return Navigator(
```

Comment:
This used to be an `Overlay`. I switched to `Navigator` as `DropDownButton` and several other material widgets require a navigator-not just an overlay (all navigators also provide overlays).

Reply:
Interesting. At first look this seems like the better approach.

Does that impact the navigator of your `*App`?
In the past I had problems with overlays not being at the correct position because I used a `MaterialApp` inside this library. MaterialApp also provides a Navigator.

An simple test for that would be to show a dialog and open the feedback view iirc.

I'll hope that this change does not reintroduce that bug.

Reply:
Ah it probably doesn't because it's just used in the FeedbackBottomSheet.

Reply:
From messing around manually with the sample app this worked well. I'm not quite sure what the right test to verify this is-but if you have ideas for test scenarios to ensure proper behavior I can try writing them up.

Reply:
One thing that could be useful: A test that opens the feedback in navigate mode, clicks a couple buttons, and then hits back a couple times and verifies that the appropriate widgets are present between each action.

Reply:
Added!